pmd-plsql/etc/grammar/PldocAST.jjt

```diff
@@ -2686,7 +2686,7 @@ ASTFetchStatement FetchStatement()  :
 ASTEmbeddedSqlStatement EmbeddedSqlStatement()  :
 {}
 {
-   <EXECUTE> <IMMEDIATE> (StringLiteral() | Expression())
+   <EXECUTE> <IMMEDIATE> Expression()
```

Comment:
Technically, this allows now more than should be allowed. Here's the doc/spec about this statement: https://docs.oracle.com/en/database/oracle/oracle-database/19/lnpls/EXECUTE-IMMEDIATE-statement.html#GUID-C3245A95-B85B-4280-A01F-12307B108DC8

It takes "dynamic_sql_stmt" as the first production, which is

> String literal, string variable, or string expression that represents a SQL statement. Its type must be either CHAR, VARCHAR2, or CLOB. 

I guess, by using "StringLiteral()" here, almost always a StringLiteral (probably something starting with `'` or `"`)  was used and almost never Expression() (only in cases a variable was used). The "||" is the [concatenation operator](https://docs.oracle.com/en/database/oracle/oracle-database/19/lnpls/plsql-language-fundamentals.html#GUID-4792284C-D82B-4F6A-8554-45CF1BFAE6EB). We use this in our production [StringExpression](https://github.com/pmd/pmd/blob/2b6dcb3f22431f511aca86d3543ecf7515892cac/pmd-plsql/etc/grammar/PldocAST.jjt#L3190-L3203).

Using Expression() here would probably resolve the "||"-operator to a AdditiveExpression which doesn't sound right.

Could you add a test case with 3 variants calling execute immediate?
1. direct string literal, e.g. `EXECUTE IMMEDIATE 'UPDATE employees_temp SET name = NULL'` (see also https://docs.oracle.com/en/database/oracle/oracle-database/19/lnpls/dynamic-sql.html#GUID-BAC52F38-B427-4860-9B6D-336B3AEE6AD2)
2. string variable, e.g. `EXECUTE IMMEDIATE plsql_block` (see also https://docs.oracle.com/en/database/oracle/oracle-database/19/lnpls/dynamic-sql.html#GUID-30ADFC05-8DFB-4C8A-831D-426C2E97F217)
3. string expression, like your example `EXECUTE IMMEDIATE 'drop database link ' || v_link`

And could you try out using here `StringExpression()` instead of `Expression()`?

Reply:
I didn't see your last comment until now. I renamed the unit test, switched to StringExpression() and move the test code to a separate file as you suggested in your comment. Regarding the test cases:
I added a test case for 3), and test cases for 1) and 2) already exist (see files resources/net/sourceforge/pmd/lang/plsql/ExecuteImmediate*.pls)